src/core/layertree/qgslayertreemodellegendnode.cpp

```diff
@@ -529,10 +529,14 @@ QVariant QgsSymbolLegendNode::data( int role ) const
           QPainter painter( &pix );
           painter.setRenderHint( QPainter::Antialiasing );
           context->setPainter( &painter );
-          const QFontMetricsF fm( mTextOnSymbolTextFormat.scaledFont( *context ) );
-          const qreal yBaselineVCenter = ( mIconSize.height() + fm.ascent() - fm.descent() ) / 2;
-          QgsTextRenderer::drawText( QPointF( mIconSize.width() / 2, yBaselineVCenter ), 0, QgsTextRenderer::AlignCenter,
-                                     QStringList() << mTextOnSymbolLabel, *context, mTextOnSymbolTextFormat );
+          bool isNullSize = false;
+          const QFontMetricsF fm( mTextOnSymbolTextFormat.scaledFont( *context, 1.0, &isNullSize ) );
```

Comment:
seeing the same pattern a bit below, shouldn't that test be moved in QgsTextRenderer::drawText() itself ?

Reply:
It's also there (in various places). The reason we also need to handle it outside of the text renderer internals is that qt will return a default sized font when we try to set the size to 0 (and there's no concept of a "null"/"invalid" qfont we can return from the function before this happens). So in this case the font metrics will incorrectly be calculated on a non zero font size and will return values for height/width etc, when we should be treating these all as 0. I can't see any alternative here.

Reply:
my point was that a 
```
bool isNullSize = false;
const QFontMetricsF fm( format.scaledFont( context, 1.0, &isNullSize ) );
if ( isNullSize )
    return;
```
at the beginning of void QgsTextRenderer::drawText( QPointF point, double rotation, QgsTextRenderer::HAlignment alignment, const QStringList &textLines, QgsRenderContext &context, const QgsTextFormat &format, bool ) would likey behave the same. But I see this would involv querying th fontmetrics twice, with perhaps performance issues, so I'm fine with the current code.